src/runtime/arc.ts

```diff
@@ -123,15 +123,7 @@ export class Arc implements ArcInterface {
       slotComposer['arc'] = this;
     }
 
-    if (typeof id === 'string') {
-      // TODO(csilvestrini): Replace this warning with an exception.
-      console.error(
-          `Arc created with string ID ${id}!!! This should be an object of type Id instead. This warning will turn into an ` +
-          `exception soon (end of April 2019).`);
-      this.id = ArcId.fromString(id);
-    } else {
-      this.id = id;
-    }
+    this.id = id;
```

Comment:
maybe, add an assert, just to be sure?

Reply:
It isn't possible, right? Since the ctor type of id is `Id`, anything that isn't that would fail to compile?

Reply:
Maybe you're saying, to be sure vs. JS, but I grepped our JS, and there isn't any, so I don't think we need the defensive code to stick around unnecessarily (but I could regret those words later).

Reply:
you're right, it doesn't seem possible.